### ISSUE TITLE
array-type doesn't consider "this" to be a simple type

### ISSUE THREAD

**Issue by stogle** [palantir/tslint#2792]
### Bug Report

- __TSLint version__: 5.2.0
- __TypeScript version__: 2.2.1
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
public children: this[];
```

with `tslint.json` configuration:

```json
{
    "extends": "tslint:recommended",
    "array-type": [true, "array-simple"]
}
```

#### Actual behavior

ERROR: test.ts[x, y]: Array type using 'T[]' is forbidden for non-simple types. Use 'Array<T>' instead.

#### Expected behavior

No warnings reported, since this should be considered a simple type along with e.g. object, number and string. See https://github.com/palantir/tslint/issues/2351

**Comment by adidahiya** [comment]
agreed, this should be a really simple PR for anyone looking to contribute!
